tests/scripts/validate_modified_files.sh

```diff
@@ -9,26 +9,31 @@ MOD_ERR="changes found by mod.check. You may need to run make clean"
 CRD_ERR="changes found by 'make crds'. please run 'make crds' locally and update your PR"
 BUILD_ERR="changes found by make build', please commit your go.sum or other changed files"
 HELM_ERR="changes found by 'make gen-rbac'. please run 'make gen-rbac' locally and update your PR"
+DOCS_ERR="changes found by 'make docs'. please run 'make docs'locally and update your PR"
+
 
 #############
 # FUNCTIONS #
 #############
 function validate(){
-  git=$(git status --porcelain)
-  for file in $git; do
-    if [ -n "$file" ]; then
-      echo "$1"
-      echo "$git"
-      git diff
+  ERR=$1
+  git diff  --exit-code
+  ret=$?
+  if [ "$ret" != "0" ]; then
+    echo "modidied files detected."
+    git status --porcelain
+    echo "$ERR"
       exit 1
-    fi
-  done
+  fi
 }
 
 ########
 # MAIN #
 ########
 case "$1" in
+docs)
+    validate "$DOCS_ERR"
```

Comment:
`echo $"Usage: $0 {codegen|modcheck|crd|build|gen-rbac}"`  should reflect `docs` as well.

Reply:
@sp98  wrote:
> `echo $"Usage: $0 {codegen|modcheck|crd|build|gen-rbac}"` should reflect `docs` as well.

Good point, thanks! Addressed via #14672  and rebase.